wallet/wallet.go

```diff
@@ -3336,7 +3338,19 @@ func (w *Wallet) reliablyPublishTransaction(tx *wire.MsgTx) (*chainhash.Hash, er
 		return nil, err
 	}
 	err = walletdb.Update(w.db, func(dbTx walletdb.ReadWriteTx) error {
-		return w.addRelevantTx(dbTx, txRec, nil)
+		if err := w.addRelevantTx(dbTx, txRec, nil); err != nil {
+			return err
+		}
+
+		// If the tx label is empty, we can return early.
+		if len(label) == 0 {
+			return nil
+		}
+
+		// If there is a label we should write, get the namespace key
+		// and record it in the tx store.
+		txmgrNs := dbTx.ReadWriteBucket(wtxmgrNamespaceKey)
+		return w.TxStore.PutTxLabel(txmgrNs, tx.TxHash(), label)
```

Comment:
Originally put this in `reliablyPublishTransaction` because it's only called for user initiated sends (where it's possible to provide a label). `addRelevantTx` is also called when we get chain notifications, so we would need to check whether there is an existing label, and only overwrite it if the label provided is not empty. Otherwise can end up with user publishing tx with label, tx confirms on chain, label overwritten when `addRelevantTx` is called called by `handleChainNotifications` with label="". 

nbd to move it down and add a check, but labels feel more high level to me, so it made sense to only write on specific use case of a user send.